src/containers/DeleteConfirmation.jsx

```diff
@@ -19,30 +20,32 @@ const DeleteConfirmation = ({ t, selected, onConfirm, onDismiss }) => {
     title={t('deleteconfirmation.title', selected.length)}
     description={deleteConfirmationTexts}
     cancelText={t('deleteconfirmation.cancel')}
-    cancelAction={() => onDismiss(selected)}
+    cancelAction={() => onDismiss(selected, dropSelection)}
     validateType='danger'
     validateText={t('deleteconfirmation.delete')}
-    validateAction={() => onConfirm(selected)}
+    validateAction={() => onConfirm(selected, dropSelection)}
    />)
 }
 
 const mapStateToProps = (state, ownProps) => {
   return {
-    selected: state.ui.selected
+    selected: getActionableFiles(state),
+    dropSelection: state.ui.selected.length > 0
   }
 }
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
-  onDismiss: (selected) => {
+  onDismiss: (selected, dropSelection) => {
     dispatch(hideDeleteConfirmation())
-    selected.forEach(item => dispatch(toggleFileSelection(item, true)))
+    if (dropSelection) selected.forEach(item => dispatch(toggleFileSelection(item.id, true)))
   },
-  onConfirm: (selected) => {
+  onConfirm: (selected, dropSelection) => {
     selected.forEach(item => {
-      dispatch(trashFile(item))
-      dispatch(toggleFileSelection(item, true))
+      dispatch(trashFile(item.id))
+      if (dropSelection) dispatch(toggleFileSelection(item.id, true))
     })
     dispatch(hideDeleteConfirmation())
+    dispatch(hideFileActionMenu())
```

Comment:
IMO, this can stay as it is for this PR, but we should probably change this later. Seeing it now, I think we are doing too many things un this component that aren't it's job (dropping the selection, hiding stuff). I think those things could be handled by reducers that react to a trash success / failure rather than explicits dispatch. Any thoughts, @goldoraf ?